src/http/test/http_client_test.cpp

```diff
@@ -32,10 +33,108 @@
 #include "utils/error_code.h"
 #include "utils/errors.h"
 #include "utils/fmt_logging.h"
+#include "utils/strings.h"
 #include "utils/test_macros.h"
 
 namespace dsn {
 
+using http_url_case =
+    std::tuple<const char *, const char *, uint16_t, const char *, const char *, const char *>;
```

Comment:
It's too long, how about adding a new struct which contains all the fields, the struct member names are self-explanatory, and it would be helpful to simplify the following code.

Reply:
Good idea, I've used `struct` for all of the cases that need many arguments.